src/PIL/JxlImagePlugin.py

```diff
@@ -0,0 +1,180 @@
+            self._tps_dur_secs = tps_num / tps_denom
+        # TODO: handle libjxl timecods
+        self.__timestamp = 0
+
+        self._mode = mode
+        self.rawmode = mode
+        self.tile = []
+
+        icc = self._decoder.get_icc()
+        exif = self._decoder.get_exif()
+        xmp = self._decoder.get_xmp()
+        if icc:
+            self.info["icc_profile"] = icc
+        if exif:
+            self.info["exif"] = self._fix_exif(exif)
+        if xmp:
+            self.info["xmp"] = xmp
+
+        self._rewind()
+
+    def _fix_exif(self, exif):
+        # jpeg xl does some weird shenanigans when storing exif
+        # it omits first 6 bytes of tiff header but adds 4 byte offset instead
+        if len(exif) <= 4:
+            return None
+        exif_start_offset = struct.unpack(">I", exif[:4])[0]
+        return exif[exif_start_offset + 4 :]
+
+    def _getexif(self):
+        if "exif" not in self.info:
+            return None
+        return self.getexif()._get_merged_dict()
+
+    def getxmp(self):
+        return self._getxmp(self.info["xmp"]) if "xmp" in self.info else {}
+
+    def _get_next(self):
+
+        # Get next frame
+        next_frame = self._decoder.get_next()
+        self.__physical_frame += 1
+
+        # this actually means EOF, errors are raised in _jxl
+        if next_frame is None:
+            msg = "failed to decode next frame in JXL file"
+            raise EOFError(msg)
+
+        data, tps_duration, is_last = next_frame
+        if is_last and self.n_frames is None:
+            # libjxl said this frame is the last one
+            self.n_frames = self.__physical_frame
+
+        # duration in miliseconds
+        duration = 1000 * tps_duration * (1 / self._tps_dur_secs)
+        timestamp = self.__timestamp
+        self.__timestamp += duration
+
+        return data, timestamp, duration, is_last
+
+    def _rewind(self, hard=False):
+        if hard:
+            self._decoder.rewind()
+        self.__physical_frame = 0
+        self.__loaded = -1
+        self.__timestamp = 0
+
+    def _seek_check(self, frame):
+        # if image is not animated then only the 0th frame is available
+        if (not self.is_animated and frame != 0) or (
+            self.n_frames is not None and (frame >= self.n_frames or frame < 0)
+        ):
+            msg = "attempt to seek outside sequence"
+            raise EOFError(msg)
+
+        return self.tell() != frame
+
+    def _seek(self, frame):
+        # print("_seek: phy: {}, fr: {}".format(self.__physical_frame, frame))
+        if frame == self.__physical_frame:
+            return  # Nothing to do
+        if frame < self.__physical_frame:
+            # also rewind libjxl decoder instance
+            self._rewind(hard=True)
+
+        while self.__physical_frame < frame:
+            self._get_next()  # Advance to the requested frame
+
+    def seek(self, frame):
+        if not self._seek_check(frame):
+            return
```

Comment:
And a test for this?

Reply:
There's already `test_file_jxl_animated.py` test which also does seeking.

Reply:
The `return` isn't covered: https://app.codecov.io/gh/python-pillow/Pillow/pull/7848/blob/src/PIL/JxlImagePlugin.py